pid_ctrl/include/pid_ctrl.h

```diff
@@ -95,6 +106,17 @@ esp_err_t pid_update_parameters(pid_ctrl_block_handle_t pid, const pid_ctrl_para
  */
 esp_err_t pid_compute(pid_ctrl_block_handle_t pid, float input_error, float *ret_result);
 
+/**
+ * @brief Reset the pre_error
+ *
+ * @param[in] pid PID control block handle, created by `pid_new_control_block()`
+ * @param[in] Intermediate_data Intermediate data that need to update
+ * @return
+ *      - ESP_OK: Run a PID compute successfully
+ *      - ESP_ERR_INVALID_ARG: Run a PID compute failed because of invalid argument
+ */
+esp_err_t pid_update_Intermediate_data(pid_ctrl_block_handle_t pid, const pid_ctrl_intermediate_data_t *Intermediate_data);
```

Comment:
Hmmm, do you think the user will know the current value of the internal "integral_err" when he only wants to reset the "previous_err" **only**? IMHO, the previous API looks better. "integral_err" doesn't have to be reset with "previous_err" together.

Reply:
Typically, this interface is only called if the user wants to zero the cumulative error, but it can't be ruled out that some users will want to set the cumulative error to try out a different control effect, but I would think that this should be very rare, and perhaps this kind of functionality shouldn't exist in the component?

Reply:
How about split the previous `reset` API into `reset_previous_err` and `reset_integral_err`? 

And is it reasonable to reset the last output together with previous error?

Reply:
Oh, maybe reset `last_output` and `integral_err` together make more sense. Because `last_output` is the accumulated variable for `incremental` formula and `integral_err` is the accumulated variable for `positional` formula. And one pid control block is attached to one formula, reset them together is reasonable. Maybe naming it `reset_accumulation`/`clear_accumulation` or something.

Reply:
the `last output` is for `PID_CAL_TYPE_INCREMENTAL` and the `integral_err` is for `PID_CAL_TYPE_POSITIONAL`. They are essentially the same thing, effective only in their respective control modes. There is no need to use multiple reset functions

Reply:
Yes, actually I mean two function like this:
```c
esp_err_t pid_reset_previous_err(pid_ctrl_block_handle_t pid)
{
    ESP_RETURN_ON_FALSE(pid, ESP_ERR_INVALID_ARG, TAG, "invalid argument");
    pid->previous_err1 = 0;
    pid->previous_err2 = 0;
    return ESP_OK;
}

esp_err_t pid_reset_accumulation(pid_ctrl_block_handle_t pid)
{
    ESP_RETURN_ON_FALSE(pid, ESP_ERR_INVALID_ARG, TAG, "invalid argument");
    pid->integral_err = 0;
    pid->last_output = 0;
    return ESP_OK;
}
```

I'm also OK with reseting all intermediate data in one function, but I think the function can be renamed to `pid_reset_ctrl_block` then.